README.md

```diff
@@ -1,2 +1,17 @@
 # go-http-middleware
 
+[![Build Status](https://drone.corp.mongodb.com/api/badges/kanopy-platform/go-http-middleware/status.svg)](https://drone.corp.mongodb.com/kanopy-platform/go-http-middleware)
+
+The `go-http-middleware` is a collection of generic HTTP middleware functions.
+
+## Design Guidelines
+
+* Middleware MUST NOT rely on external application specific state
+* Middleware MAY add information to the request context for downstream use.
+* Generic Middleware MUST not interrupt the chain. It MUST always call the next handler.
```

Comment:
The meaning of this is a bit vague. Middleware can have failure cases that would interrupt the chain and return failure codes.

Reply:
I can see an argument for that.  I will remove this.